<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>FilterDialog</class>
 <widget class="QDialog" name="FilterDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>528</width>
    <height>485</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Configure filter</string>
  </property>
  <property name="windowIcon">
   <iconset theme="view-filter">
    <normaloff>.</normaloff>.</iconset>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="6" column="0">
    <widget class="QRadioButton" name="radioButtonCustomTaps">
     <property name="text">
      <string>Custom taps:</string>
     </property>
    </widget>
   </item>
   <item row="11" column="0">
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>40</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="5" column="0" colspan="2">
    <widget class="QRadioButton" name="radioButtonMovingAverage">
     <property name="text">
      <string>Moving average filter</string>
     </property>
     <property name="checked">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="10" column="0" colspan="2">
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
     <property name="centerButtons">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="0" column="0" colspan="2">
    <widget class="QLabel" name="label_3">
     <property name="font">
      <font>
       <italic>true</italic>
      </font>
     </property>
     <property name="text">
      <string>You can imagine taps as weighting factors applied to n samples of the signal whereby n is the number of taps.  By default we use 10 taps with each tap set to 0.1 producing a moving average filter.</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="1" column="0" colspan="2">
    <widget class="QLabel" name="label_2">
     <property name="font">
      <font>
       <italic>true</italic>
      </font>
     </property>
     <property name="text">
      <string>These n weighted samples get summed up to produce the output of the filter. In DSP terms you configure the impulse response of the filter here.</string>
     </property>
     <property name="wordWrap">
      <bool>true</bool>
     </property>
    </widget>
   </item>
   <item row="6" column="1">
    <widget class="QLineEdit" name="lineEditCustomTaps">
     <property name="toolTip">
      <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;You can configure custom filter taps here either explicit using [0.1, 0.2, 0.3] or with &lt;span style=&quot; font-weight:600;&quot;&gt;python programming shortcuts&lt;/span&gt; like [0.1] * 3 + [0.2] * 4 will result in [0.1, 0.1, 0.1, 0.2, 0.2, 0.2, 0.2]&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
     </property>
     <property name="text">
      <string>[0.1]*10</string>
     </property>
    </widget>
   </item>
   <item row="7" column="0">
    <widget class="QLabel" name="label">
     <property name="text">
      <string>Number of taps:</string>
     </property>
    </widget>
   </item>
   <item row="7" column="1">
    <widget class="QSpinBox" name="spinBoxNumTaps">
     <property name="minimum">
      <number>1</number>
     </property>
     <property name="maximum">
      <number>999999999</number>
     </property>
     <property name="value">
      <number>10</number>
     </property>
    </widget>
   </item>
   <item row="2" column="0" colspan="2">
    <widget class="QRadioButton" name="radioButtonDCcorrection">
     <property name="text">
      <string>DC correction</string>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
